aicssegmentation/workflow/workflow.py

```diff
@@ -162,3 +169,148 @@ def is_done(self) -> bool:
+        """
+        Check if file at a given image_path and is a valid image type we support.
+
+        Params:
+            image_path (Path): image to check
+
+        Returns:
+            (bool): True if file has a supported file extension.
+        """
+        if not image_path.exists():
+            return False
+        if image_path.suffix.lower() in SUPPORTED_FILE_EXTENSIONS:
+            return True
+        else:
+            return False
+
+    def process_all(self):
+        """
+        Process all images in the input_dir with the workflow_definition used to set up the BatchWorkflow
+
+        Params:
+            none
+
+        Returns:
+            none
+        """
+        files = [f for f in self.input_path.glob("**/*") if f.is_file]
+        # Currently will save files in same format as they are in the input path
+        for f in files:
+            self._files_count += 1
+            if self.is_valid_image(f):
+                read_image = AICSImage(f)
+                # read and format image in the way we expect
+                image_from_path = self.format_image_to_3d(read_image)
+                try:
+                    # Run workflow on image
+                    workflow = Workflow(self._workflow_definition, image_from_path)
+                    result = workflow.execute_all()
+                    with OmeTiffWriter(self.output_path.joinpath(f.name), overwrite_file=True) as w:
+                        w.save(data=self.convert_bool_to_uint8(result), dimension_order="ZYX")
+
+                except Exception as e:
+                    # Handle failures during workflow execution/save
+                    self._failed_files += 1
+                    with open(self._log_file, "w") as log:
+                        log.write(f"FAILED: {f}, ERROR: {e}")
+            else:
+                self._failed_files += 1
+                with open(self._log_file, "w") as log:
+                    log.write(f"FAILED: {f}, ERROR: Unsupported Image Type {f.suffix}")
+        self._write_log_file_summary()
+
+    def _write_log_file_summary(self):
+        """
+        Write a log file to the output folder.
+
+        Params:
+            none
+
+        Returns:
+            none
+        """
+        with open(self._log_file, "w") as f:
+            if self._files_count == 0:
+                f.write("There were no files to process in the input directory")
+            else:
+                files_processed = self._files_count - self._failed_files
+                f.write(f"{files_processed}/{self._files_count} files were processed.\n")
+
+    def format_image_to_3d(self, image: AICSImage) -> np.ndarray:
+        """
+        Format images in the way that aics-segmention expects for most workflows (3d, zyx)
+
+        Params:
+            image_path (AICSImage): image to format
+
+        Returns:
+            np.ndarray: segment-able image for aics-segmentation
+        """
+        if len(image.shape) == 6:
```

Comment:
i think we may need anothe check here. For example, the .czi file is a multi-scene timelapse, the current batch processing only works on scene 0 and time 0. So, we should warn users that this is the case. In future versions, we could process all individual scene and time, but no worry for the moment. Just a warning should be enough

Reply:
Should the warning bring up a warning box? Or should it just set S and T to 0 and write to the log file @jxchen01 ?

Reply:
on the log file should be fine. If possible, I would suggest to skip such files and report the error in the log when a multi-scene or timelapse is encountered. We can leave all such scenarios to future.